datajunction-server/datajunction_server/api/metrics.py

```diff
@@ -69,17 +71,23 @@ async def list_metrics(
     validate_access: access.ValidateAccessFn = Depends(
         validate_access,
     ),
+    cache: Cache = Depends(get_cache),
+    background_tasks: BackgroundTasks,
 ) -> List[str]:
     """
     List all available metrics.
     """
-    return await list_nodes(
-        node_type=NodeType.METRIC,
-        prefix=prefix,
-        session=session,
-        current_user=current_user,
-        validate_access=validate_access,
-    )
+    metrics = cache.get("metrics")
```

Comment:
Can we set this up so that the `cache` here is optional? And thus if it's configured, we can check against it, but if it's not configured, we'll just fallback to the regular behavior.

Reply:
Ah never mind, I see that this would fallback to a `NoOpCache`. LGTM then!